src/FsAutoComplete.Core/Commands.fs

```diff
@@ -248,150 +257,158 @@ type Commands (serialize : Serializer) =
-    }
-
-    member __.GetUnionPatternMatchCases (tyRes : ParseAndCheckResults) (pos: Pos) (lines: LineStr[]) (line: LineStr) = async {
-        let codeGenService = CodeGenerationService(checker, state)
-        let doc = {
-            Document.LineCount = lines.Length
-            FullName = tyRes.FileName
-            GetText = fun _ -> lines |> String.concat "\n"
-            GetLineText0 = fun i -> lines.[i]
-            GetLineText1 = fun i -> lines.[i - 1]
-        }
-
-        let! res = tryFindUnionDefinitionFromPos codeGenService pos doc
-        match res with
-        | None -> return [Response.info serialize "Union at position not found"]
-        | Some (symbolRange, patMatchExpr, unionTypeDefinition, insertionPos) ->
-
-        if shouldGenerateUnionPatternMatchCases patMatchExpr unionTypeDefinition then
-            let result = formatMatchExpr insertionPos "$1" patMatchExpr unionTypeDefinition
-            let pos = {
-                Col = insertionPos.InsertionPos.Column
-                Line = insertionPos.InsertionPos.Line
+        async {
+            let res =
+                match state.TryGetFileCheckerOptionsWithSource file with
+                | Failure s -> [Response.error serialize s]
+                | Success (options, source) ->
+                    let tyResOpt = checker.TryGetRecentCheckResultsForFile(file, options)
+
+                    match tyResOpt with
+                    | None -> [ Response.info serialize "Cached typecheck results not yet available"]
+                    | Some tyRes ->
+                        match tyRes.GetAST with
+                        | None -> [ Response.info serialize "Something went wrong during parsing"]
+                        | Some tree ->
+                            try
+                                fsharpLintConfig.LoadConfigurationForProject file
+                                let opts = fsharpLintConfig.GetConfigurationForProject (file)
+                                let res =
+                                    Lint.lintParsedSource
+                                        { Lint.OptionalLintParameters.Default with Configuration = Some opts}
+                                        { Ast = tree
+                                          Source = source
+                                          TypeCheckResults = Some tyRes.GetCheckResults
+                                          FSharpVersion = Version() }
+                                let res' =
+                                    match res with
+                                    | LintResult.Failure _ -> [ Response.info serialize "Something went wrong, linter failed"]
+                                    | LintResult.Success warnings -> [ Response.lint serialize warnings ]
+
+                                res'
+                            with _ex ->
+                                [ Response.info serialize "Something went wrong during linter"]
```

Comment:
Not a huge fan of throwing error information away, but there are other places and it is not added here.